larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -460,6 +460,7 @@ void ThreeDVertexDistanceFeatureTool::Run(LArMvaHelper::MvaFeatureVector &featur
     if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
         std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
 
+    LArMvaHelper::MvaFeature vertexDistance;
     const VertexList *pVertexList(nullptr);
     (void) PandoraContentApi::GetCurrentList(*pAlgorithm, pVertexList);
 
```

Comment:
What do you think about i) checking whether pVertexLists is not nullptr here, and ii) checking to see whether the list isn't empty, up front, here? Because can't calculate vertexDistance below if there's no vertex. Could then remove some vertex checks below, but that might actually make implementation less robust against future changes.

Reply:
Good shout, doing the sanity checks earlier is more in keeping with the other pandora code snippets I've seen.
I've added this in.
One point, I've slightly broken the style of the function ie. the vertex list sanity check fills the feature vector and returns.  However, the more complicated checks still only set `vertexDistance` and then wait for the final `push_back` at the end of the function.

Reply:
Currently, in the case of an empty vertex list the returned distance to the vertex is zero, based on discussion elsewhere I think the conclusion was that a std::numeric_limits::max variant was preferred, is that still the case?

Reply:
In the update above to get the MvaFeature type consistency for the vertex distance, the value used has been reset to zero in the case where the vertex list is empty, which, from discussion elsewhere, I think is not what is wanted.

Reply:
H @AndyChappell 
My intention was to return a default-initialised `vertexDistance` when the actual vertex distance couldn't be calculated, which I think is what the original code tried to do.  Is that not the case?

Reply:
Ah I saw the email notification for your other comment (but cannot find it here bizarrely).
The very original plan was to return an extreme value for a default but that was before I knew that the type was wrapped in an initialisation bool.  So, I opted to preserve how the original code handled defaults.